test/controllers/api/v2/registration_controller_test.rb

```diff
@@ -69,6 +70,47 @@ class Api::V2::RegistrationControllerTest < ActionController::TestCase
       assert_response :not_found
       assert_includes @response.body, "echo \"Couldn't find Hostgroup with 'id'=0\""
     end
+
+    context 'with :url parameter' do
+      test 'without protocol and without port' do
+        get :global, params: { url: 'example.com' }, session: set_session_user
+        assert_response :internal_server_error
+      end
+
+      test 'without protocol and with port' do
+        get :global, params: { url: 'example.com:0' }, session: set_session_user
+        assert_response :internal_server_error
+      end
+
+      test 'with protocol' do
+        url = 'https://example.com'
+        get :global, params: { url: url }, session: set_session_user
+        assert_response :success
+        assert_equal "#{url}/register", assigns(:global_registration_vars)[:registration_url].to_s
+      end
+
+      test 'with port' do
+        url = 'https://example.com:0'
+        get :global, params: { url: url }, session: set_session_user
+        assert_response :success
+        assert_equal "#{url}/register", assigns(:global_registration_vars)[:registration_url].to_s
+      end
+
+      test 'with path' do
+        url = 'https://example.com/this-path-should-not-be-here'
+        get :global, params: { url: url }, session: set_session_user
+        assert_response :success
+        assert_equal 'https://example.com/register', assigns(:global_registration_vars)[:registration_url].to_s
+      end
+
+      test 'with RAILS_RELATIVE_URL_ROOT' do
+        ENV['RAILS_RELATIVE_URL_ROOT'] = '/foreman'
```

Comment:
we should reset the ENV variable to previous value at the end of the test, this could cause weird problems in other tests I think